src/utils/FileEmitter.ts

```diff
@@ -318,17 +318,27 @@ export class FileEmitter {
 		// We must not modify the original assets to avoid interaction between outputs
 		const assetWithFileName = { ...consumedFile, source, fileName };
 		this.filesByReferenceId.set(referenceId, assetWithFileName);
-		output.bundle[fileName] = { fileName, isAsset: true, source };
+		const self = this;
```

Comment:
I guess this is really personal taste and I know why you did this but I always have to cringe when I see this pattern. How about instead you do `const graph = this.graph` and then `graph.warnDeprecation(..)` later? Then it is also immediately obivous why you do this here, and we could even save a property access if we need `graph` for other things in this method at some point in the future.